README.md

```diff
@@ -1,11 +1,130 @@
+
+* Next, install the required dependencies:
+
+```
+npm install
+```
+
+* First, run CDK bootstrap using the AWS account ID and region you specified in the `cdk.context.json` file. 
+* You need to provide all the required context parameters in the command
+
+```
+cdk bootstrap aws://<aws-account-number>/<aws-region> --context securityDisabled=true \
+--context minDistribution=false --context distributionUrl='https://artifacts.opensearch.org/releases/bundle/opensearch/2.3.0/opensearch-2.3.0-linux-x64.tar.gz' \
+--context cpuArch='x64' --context singleNodeCluster=false --context dataNodeCount=3 \
+--context dashboardsUrl='https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.3.0/opensearch-dashboards-2.3.0-linux-x64.tar.gz' \
+--context distVersion=2.3.0
+```
+* Now you are ready to synthesize the CloudFormation templates:
+
+```
+cdk synth "* " --context securityDisabled=true \
+--context minDistribution=false --context distributionUrl='https://artifacts.opensearch.org/releases/bundle/opensearch/2.3.0/opensearch-2.3.0-linux-x64.tar.gz' \
+--context cpuArch='x64' --context singleNodeCluster=false --context dataNodeCount=3 \
+--context dashboardsUrl='https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.3.0/opensearch-dashboards-2.3.0-linux-x64.tar.gz' \
+--context distVersion=2.3.0
+```
+
+#### Sample command to set up multi-node cluster with security disabled on x64 AL2 machine
+
+Please note that as of now we only support instances backed by Amazon Linux-2 amis.
+
+```
+cdk deploy "*" --context securityDisabled=true \
+--context minDistribution=false --context distributionUrl='https://artifacts.opensearch.org/releases/bundle/opensearch/2.3.0/opensearch-2.3.0-linux-x64.tar.gz' \
+--context cpuArch='x64' --context singleNodeCluster=false --context dataNodeCount=3 \
+--context dashboardsUrl='https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.3.0/opensearch-dashboards-2.3.0-linux-x64.tar.gz' \
+--context distVersion=2.3.0
+```
+
+### Interacting with OpenSearch cluster
+
+After CDK Stack deployment the user will be returned a load-balancer url which they can use to interact with the cluster.
+
+#### Sample commands
+`curl -X GET "http://<load-balancer-url>/_cluster/health?pretty"` for OpenSearch
+
+To interact with dashboards use port `8443`. Type `http://<load-balancer-url>:8443` in your browser.
+
+For security enabled cluster run `curl -X GET https://<load-balancer-url> -u 'admin:admin' --insecure`
+The security enabled dashboard is accessible using `http` on port `8443`
+
+#### Please note the load-balancer url is internet facing and can be accessed by anyone.
+To restrict access please refer [Client IP Preservation](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html#client-ip-preservation) to restrict access on internet-facing network load balancer.
+You need to add the ip/prefix-list rule in the security group created in network stack.
```

Comment:
Can we make this as a required parameter from the user? Either provide a`securityGroupId` or the access list. Example to add that https://github.com/opensearch-project/opensearch-ci/pull/229

Since the cluster will be open to all by default and can contain customer data we need to enforce security from our end.

Reply:
Can you replace this section with restricting server access one? I believe you posted that somewhere else?